wscript

```diff
@@ -336,6 +336,8 @@ but you then may not have a local copy of the HTML manual.'''
     conf.env.append_value('CFLAGS', ['-DGTK'])
     conf.env.append_value('CXXFLAGS',
         ['-DNDEBUG', '-DGTK', '-DSCI_LEXER', '-DG_THREADS_IMPL_NONE'])
+    if conf.env['CXX_NAME'] == 'gcc' and '-O' not in ''.join(conf.env['CXXFLAGS']):
```

Comment:
`gcc` is actually correct, at least in the Waf world.
See https://github.com/waf-project/waf/blob/waf-1.7.16/waflib/Tools/gxx.py#L22.
Not sure why Waf sets CXX_NAME to gcc and not g++, maybe they refer to the GNU Compiler Collection as the whole software suite instead of the specific binary name.
Anyway, in this case it should be safe to assume `gcc`.

Regarding other possible compilers: no idea.
In the past, we tried MSVC but without success. I don't remember the details but Geany could not be successfully built with MSVC. Maybe this will change in the future. Currently, we actually have a hack in our wscript to disable MSVC detection on Windows.
Also, we played with suncc at some point in the past. As above, I don't remember the details. Shame on my memories :(.